lib/facility.rb

```diff
@@ -1,14 +1,65 @@
+require 'date'
+
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_accessor :name, :address, :phone, :services, :registered_vehicles, :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(facility_info)
+    @name = facility_info[:name]
+    @address = facility_info[:address]
+    @phone = facility_info[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle)
+    return nil if !@services.include?("Vehicle Registration")
+    if @services.include?("Vehicle Registration")
+      @registered_vehicles << vehicle
+      vehicle.registration_date = Date.today
+      if vehicle.plate_type == :antique
+        @collected_fees += 25
+        elsif vehicle.plate_type == :ev
+        @collected_fees += 200
+        elsif vehicle.plate_type == :regular
+        @collected_fees += 100
+      end
+    end
+    @registered_vehicles
+  end
+
+  def collected_fees
+    @collected_fees
+  end
+
+  def administer_written_test(registrant)
+    if @services.include?("Written Test") && registrant.permit? == true && registrant.age >= 16
```

Comment:
I think abstracting some of these long lines to a helper method could be more readable.
```
return false if !@services.include?("Written Test")
if written_eligible?(registrant)
```
```
def written_eligible?(registrant)
  registrant.permit? && registrant.age >= 16
end
```
Note that we can omit `== true` from this because `registrant.permit?` will already return a boolean value.